index.js

```diff
@@ -0,0 +1,209 @@
+const typeScriptSettings = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true,
+    },
+    sourceType: 'module',
+    project: './tsconfig.json',
+  },
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
+  ],
+  rules: {
+    '@typescript-eslint/array-type': ['error', { default: 'generic', readonly: 'generic' }],
+    '@typescript-eslint/ban-ts-comment': 'error',
+    '@typescript-eslint/consistent-type-assertions': 'error',
+    '@typescript-eslint/explicit-member-accessibility': 'error',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/naming-convention': [
```

Comment:
Another use case is when working with react components. They only work if they are PascalCase.

Reply:
That's already been overwritten for jsx/tsx files specifically.

Reply:
I agree with @ThijsTyZ on keeping the rule and disabling as needed. Even for React it is not too common to run into having to assign to a PascalCase variable to use in JSX.

Reply:
I've seen it used some more in dynamic/composable components where you pass components as configuration. But if that's the case, it can be disable in that file, or even the single project if needed.

Just wanted to check :)